### ISSUE TITLE
XRDP and VNC inconsistencies

### ISSUE THREAD

**Issue by SensorAnalyticsAus** [raspberrypi/bookworm-feedback#269]
I have RPI Imager installs of Bookworm on a PI5 and a PI3B. Both also have `xrdp` installed, *VNC* enabled, and *W1 X11     Openbox window manager with X11 backend* option selected within `raspi-config`. PI5 can be connected over `xrdp` while *VNC* comes up with *Cannot currently show the desktop* after successfully authenticating to *VNC*. 

OTH PI3B connection over `xrdp` fails with MS Remote Desktop client opening up a black screen which after a few minutes timesout to a grey popup ending with a message *error connecting to user session* with an *OK* button. Clicking on it leads to another popup screen for re-typing in the password, and this also fails after a long wait. Last few entries in `.xorgxrdp-11.log`:
```
[867399.538] (WW) xf86CloseConsole: KDSETMODE failed: Bad file descriptor
[867399.538] (WW) xf86CloseConsole: VT_GETMODE failed: Bad file descriptor
[867399.541] (EE) Server terminated with error (1). Closing log file.
```
paradoxically, *VNC* connection works on PI3B.

#### RPIs OS and HW info
PI5 
`uname -a
Linux rpi5 6.6.31+rpt-rpi-2712 #1 SMP PREEMPT Debian 1:6.6.31-1+rpt1 (2024-05-29) aarch64 GNU/Linux`
```
cat /proc/cpuinfo
...
Revision	: d04170
Serial		: c4c6119e310d22ba
Model		: Raspberry Pi 5 Model B Rev 1.0
```

PI3B 
`uname -a
Linux rpi3bw 6.6.31+rpt-rpi-v8 #1 SMP PREEMPT Debian 1:6.6.31-1+rpt1 (2024-05-29) aarch64 GNU/Linux`
```
cat /proc/cpuinfo
...
Revision	: a22082
Serial		: 00000000a37635af
Model		: Raspberry Pi 3 Model B Rev 1.2
```

I can get to gui desktop on both these headless PI's (one way or the other) but these inconsistencies are not re-assuring.

**Comment by lurch** [comment]
Using the X11 backend is deprecated on Raspberry Pi 5. Are you able to use the Wayfire backend and WayVNC instead?

**Comment by SensorAnalyticsAus** [comment]
I have switched to Wayfire backend on PI5, no change i.e. xrdp works and vnc shows 'cannot currently show the desktop' message. How does one use WayVNC? Can't see it in `raspi-config` only option is to enable VNC (which is done).

**Comment by lurch** [comment]
https://www.raspberrypi.com/documentation/computers/remote-access.html#vnc

I believe that if you use `raspi-config` to enable VNC, it'll automatically use WayVNC if you're using Wayfire or vncserver-x11 if you're using X11. So if you switched to X11 and then enabled VNC, and then switched back to Wayfire, _perhaps_ it's still trying to use vncserver-x11 with Wayfire? (which obviously won't work).
Does it work if (after you've switched to Wayfire) you use raspi-config to disable VNC (i.e. disable vncserver-x11), reboot, and then use raspi-config to enable VNC again (i.e. enable WayVNC)?

Note that when using WayVNC you _might_ also need to use a different VNC client to what you were using with vncserver-x11 - the documentation link I provided above recommends TigerVNC.

**Comment by SensorAnalyticsAus** [comment]
I was able to switch to WayVNC with your docu link and  information, thanks. However tigerVNC client (1.3.1 for MacOS) connects to a blank grey screen. 
![Screenshot 2024-07-03 at 12 39 32 am](https://github.com/raspberrypi/bookworm-feedback/assets/114129496/75e51a86-3a67-4dad-9a2e-4efa352cb292)

**Comment by lurch** [comment]
Works fine for me with TigerVNC Viewer v1.13.1 on Linux. I'll check with one of my Mac-using colleagues when he's back in the office next week. Did you also mean 1.13.1, or are you actually using [1.3.1](https://github.com/TigerVNC/tigervnc/releases/tag/v1.3.1) from June 2014? :wink:

**Comment by SensorAnalyticsAus** [comment]
I've indeed installed TigerVNC Viewer v1.13.1, as you guessed precisely, on  intel macos 13.6.7 Thanks, that'll be good to know.

**Comment by spl237** [comment]
Just tried TigerVNC 1.13.1 on my Mac running Ventura 13.6.7 - works fine for me with VNC on the latest image.

**Comment by lurch** [comment]
@SensorAnalyticsAus Can you try testing this again please, using a freshly-written Bookworm image on your Pi 5?

**Comment by SensorAnalyticsAus** [comment]
I was thinking of doing the same, have ordered a sdcard.

**Comment by SensorAnalyticsAus** [comment]
> @SensorAnalyticsAus Can you try testing this again please, using a freshly-written Bookworm image on your Pi 5?

Done a fresh install of Bookworm 64bit with Desktop (Recommended ver.) on a new sdcard with latest PI Imager for PI5. Both VNC and XRDP connections are working now 👍  Only one minor bug encountered, the *Shutdown* *Reboot* *Logout* dialog box doesn't work within XRDP. It does work within VNC though. Anyway, a good outcome; this glitch's immaterial.  Thanks

**Comment by andys73** [comment]
I am using a bookworm Desktop 64 bit fresh install on a Pi3B .

I get the same via XRDP

[  5368.877] (WW) xf86CloseConsole: KDSETMODE failed: Bad file descriptor
[  5368.877] (WW) xf86CloseConsole: VT_GETMODE failed: Bad file descriptor
[  5368.879] (EE) Server terminated with error (1). Closing log file.

Via VNC I get a blank screen using latest tiger VNC 1.14
![image](https://github.com/user-attachments/assets/d5bfaca2-1806-47a0-bc38-9c51796d2c1c)

**Comment by andys73** [comment]
actually I had pi booting up in console mode... changed to desktop and now I am getting this via VNC
![image](https://github.com/user-attachments/assets/9b8e8c65-bde5-401c-b89c-b99119fa4421)

**Comment by andys73** [comment]
this was solved by downgrating to tinyvnc 1.13.1...

RDP still does not work but I can now remotely connect to my bookworm pi :D

**Comment by lurch** [comment]
Okay, closing this issue.

**Comment by asfasty** [comment]
ok, after spending days to make xrdp work which definitely is working on the bulleseye edition I do not understand why this is closed. Can you please elaborate why? I cannot repeat all the config files I have tried - but I can definitely say - rdp from windows works with pre-bookworm and I just do not get it - why. My edge tabs are showing around 30+ tabs of what I have tried but I get the impression rdp from Windows is not wanted. If so I am  happy to be told so in order to use another app to access devices which ususally have been accessible by using rdcman

**Comment by spl237** [comment]
Bullseye uses X11 as its windowing system. Bookworm uses Wayland, which is a completely different windowing system. RDP accesses the windowing system, so it looks as if while RDP was compatible with X11, it is not compatible with Wayland.

If you wish to remote access a Bookworm Pi, use VNC - there is a version of VNC which is compatible with Wayland, and this is already installed on Bookworm.

**Comment by asfasty** [comment]
Thank you very much for this detailed explanation. I could not find this anywhere else explained so straightforward. 👍 :)

**Comment by ghollingworth** [comment]
Or just use Pi Connect, since it "Just Works ^tm"

**Comment by asfasty** [comment]
Thank you for the additional hint. I have a pi 3 b (no plus) with touchpad - and whatever I try and what I learned is that with the older os versions the chromium was eating it up (unresponive or laggy) - now I started again to to see if a newer distro can help, then found that my usual headless doesn't work. and only from pi 4 (400 etc.) the PiConnect is possible. For me just wanting to get my small heating surveillance in place that is too much, also the time i spent on it already. The touch could be powered extra - but that did not help with the 'desktop' chromium problem I had - now using pi apps and trying to use a different light weight browser I am still stuck, this thing with touch original from raspbi is useless. Even just auto starting a page to homeassistant did end up in a stall, so therefore so much I dislike to do that - afraid it goes to the bin... - I am using stuff until it cannot be used anymore usually, repair, whatever, but this thing has cost me so much in time and effort - it is just stupid...and not supported anymore from what I understand... - Anyways thank you for picking up on my questions I appreciate that very much.
